wordpress-mobile/WordPress-iOS
# Reader post details: add ability to mark as seen/unseen.

Ref #15355 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/327

This adds the ability to mark a post seen/unseen from post details. (The same functionality will be added to the Reader cards in a follow-up PR.)

  - The `isSeen` property is now parsed and added to `ReaderPost`.
  - That property is used to determine which option to display in the menu (mark as seen/ mark as unseen).
  - When the menu option is selected, the corresponding endpoint is called to update the post's seen status.

To test:
- Go to the Reader.
- Tap on a post.
- In a browser, view the same post in the Reader.
- In the app, tap the menu button on the nav bar.
- On the menu, verify the seen/unseen option matches that in the Reader.

| ![app_mark_seen](https://user-images.githubusercontent.com/1816888/104630867-9e6d3500-5658-11eb-9656-9dc3e5e63fed.jpg) | <img width="253" alt="web_mark_seen" src="https://user-images.githubusercontent.com/1816888/104630783-7978c200-5658-11eb-8d28-9a21e5bf0645.png"> |
|--------|-------|

- Select the menu option.
- Redisplay the menu. Verify the option is now the opposite.
- Refresh the post in the browser. Verify the option is now the opposite.
  - NOTE: Sometimes it take a minute/a few refreshes for the status to be updated.

| ![app_mark_unseen](https://user-images.githubusercontent.com/1816888/104630955-c197e480-5658-11eb-8c5c-b71bd7f5ae82.jpg) | <img width="262" alt="web_mark_unseen" src="https://user-images.githubusercontent.com/1816888/104630977-c8265c00-5658-11eb-8874-0e1cb6427c0f.png"> |
|--------|-------|

PR submission checklist:
